@lingosta/app/hooks/useUser.ts

```diff
@@ -0,0 +1,58 @@
+import { createContext, useContext, useState, useEffect } from 'react';
+import { account } from '../utils/appwrite';
+import { useRouter } from 'next/navigation';
+import { toast } from 'react-hot-toast';
+
+const useUser = () => {
+
+  const router = useRouter();
+  const [user, setUser] = useState(null);
+
+  const checkUser = async() => {
+    try {
+      const response = await account.get();
+      setUser(response)
+      console.log('user exists')
+    } catch(error) {
+      console.log(error)
```

Comment:
Let's raise the error. `throw e;`

It's useful to have it logged, but we don't want our application silently fail. This is often called "swallowing an exception". If someone passes in a wrong password, how will you know that there was a failure? Keep the log but also let the error bubble up so that the appropriate UI code can render it.